src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/ApplicationsController.kt

```diff
@@ -134,15 +136,29 @@ class ApplicationsController(
     return ResponseEntity.ok(getPersonDetailAndTransform(updatedApplication, user))
   }
 
-  private fun getPersonDetailAndTransformToSummary(
-    application: uk.gov.justice.digital
-    .hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationSummary,
-    user: NomisUserEntity,
-  ):
-    ApplicationSummary {
-    val personInfo = offenderService.getInfoForPersonOrThrowInternalServerError(application.getCrn())
+  private fun getPeopleDetailsAndTransformToSummaries(applications: List<Cas2ApplicationSummary>, user: NomisUserEntity): List<ApplicationSummary> {
+    val peopleInfo = offenderService.getOffenderSummariesByCrns(
+      applications.map { it.getCrn() }.toSet(),
+    )
+
+    val summaries = mutableListOf<ApplicationSummary>()
+
+    for (application in applications) {
+      val crn = application.getCrn()
+      val personInfo = when (val offenderDetailSummary = peopleInfo[crn]) {
+        null -> PersonInfoResult.Success.Restricted(crn, application.getNomsNumber())
+        else -> offenderService.getInfoForOffender(application.getCrn(), offenderDetailSummary)
+      }
+
+      summaries.add(
+        applicationsTransformer.transformJpaSummaryToSummary(
```

Comment:
I think this could be simplified by re-using what we have elsewhere, and just passing in the person name to the transformer because the whole PersonInfoResult is not needed except for the name?
Did you take at what we've done in the Submissions Controller? Could we  follow the same pattern and re-use `getMapOfPersonNamesAndCrns` from the OffenderService?

https://github.com/ministryofjustice/hmpps-approved-premises-api/blob/3c8ba93099d7f5e0f25efb253dedc78c78f4ac48/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/SubmissionsController.kt#L181

Reply:
Ah but that would break the frontend wouldn't it...